---
title: <center>Some Pedigrees</center>
author: <center>Bruce Cochrane</center>
date: <center>`r format(Sys.Date(),format="%B %d %Y")`</center>
output: 
    html_document:
       css: Chapter.css
       toc: no
       toc_depth: 5
---

```{r,echo=FALSE}
# some housekeeping to attend to:
library(kinship2)
library(randomNames)
drawped <-function(nm,cx=1){
  file.nm <-paste("./scratch/",nm,".csv",sep="")
  dat <-read.csv(file.nm)
  ped <-with(dat,pedigree(id,dadid=fatherid,momid=motherid,sex=sex,affected=aff,missid="0"))
plot(ped,cex=cx,keep.par=TRUE)
# randomNames(40,which.names="first",gender=0,ethnicity=5)
}
# note that all files are in path ./scratch
```

### Background

So in what follows, we're going to look at a few pedigrees.  In each case, our job is to determine the *most likely* mode of inheritance - dominant or recessive; autosomal or X-linked.  Note that in many real cases, the amount of information available may be insufficient to come to a definitive conclusion

### Waardenberg Syndrome

The pedigree below is a real one, of a family in which Waardenberg's syndrome is common.  Look at it and ask the following:

1.  What is the most likely mode of inheritance?
2.  What evidence in the pedigree led you to your conclusion?

```{r}
drawped("WS3",.8)
```
Note that in this pedigree, standard notation for each individual is used.  Each person is specified by two numbers, the first specifying in which generation they occur (in this case 1 through 4) and the second one an arbitrary number from 1 through n, where n is the number of individuals in the family

### British Royal Family

Now we look at another real case, that of hemophilia in  Queen Victoria and her descendants.  In looking at this one, consider the same two questions.  Furthermore, what might you speculate about Victoria herself, given that she had an affected child *but* there is no record of the disease's occurrence in the older royal pedigrees.
```{r}
drawped("Royal2",.6)
```


### Problem 1

Now we turn to some situations we might imagine could occur, and you as a genetic counselor might be called upon to provide feedback to prospective parents.  In this case, Alena and Tyler are first cousins who have married and are thinking about having children.  Once again, address the two questions we started with, only now what we also have to do is to answer the following questions from Alena and Tyler:

1.  What is the probability that we will have an affected child?
2.  Does the sex of the child matter in terms of that probability (note that it's drawn as a male, but in fact could be either a male or a female)?

```{r}
drawped("Prob1")
```

### Problem 2

This one is similar to problem 1, only there is no consanguinity involved.  You should look at it and think about it in the same way you did for problem 1.

```{r}
drawped("Prob2",.8)
```

Now you have to do the same as you did in problem 1 with respect to Skylar and Kiana's decision-making about having children.

### Problem 3

Look at the following simple pedigree - what can you hypothesize about the mode of inheritance?

```{r,echo=FALSE}
dat.ip <-read.csv("./scratch/IncPenetrance.csv")
dat.ip
#plot pedigree 1
dat.f1 <-dat.ip[1:5,]
ped <-with(dat.f1,pedigree(id,dadid=fatherid,momid=motherid,sex=sex,affected=aff,missid="0"))
plot(ped)
```
But on some further exploration, it turns out that Cameron and Bella had another son, William, who was unaffected.  When his side of the family is added, we see the following:
```{r,echo=FALSE}
drawped("IncPenetrance")
```
How does this affect your thinking?  On the left side of the pedigree, the rules for autosomal dominant appear to hold - occurring in both sexes, with every affected individual having an affected parent.  However, on the right side it breaks down - Jevon and Alicia are affected children of unaffected parents, suggesting an autosomal recessive.  Assuming only one gene is involved, how can we explain this?  There are a couple of possibilities:

1.  If Cameron and Leah and Crystal are in fact heterozygous, then the pedigree from result from a recessive autosomal allele.  But if the condition is rare, the probability of that is negligible.
2.  This could be an example of **incomplete penetrance**, such that the allele is an autosomal dominant, however for whatever reason the trait did not manifest itself in William.  If he were *at the genotypic level* heterozygous for such an allele, then the rules for such a trait would apply.


### The Habsburgs

We will build the pedigree with the function:

```{r}
drawped("Habsburg")
```
We can also determine the kinship matrix
```{r}
dat.hb <-read.csv("./scratch/Habsburg.csv")
ped.hb <-with(dat.hb,pedigree(id,dadid=fatherid,momid=motherid,sex=sex,affected=aff,missid="0"))
kin.hb <-kinship2::kinship(ped.hb)

```
OK, now the fun thing will be to extract the diagonal
```{r}
i <-c(1:nrow(kin.hb))
data.frame(row.names(kin.hb),sapply(i,function(x) kin.hb[x,x]))
```
```{r,eval=FALSE}
source("http://bioconductor.org/biocLite.R")
biocLite()
biocLite("GeneticsPed")
```
```{r,eval=FALSE}
library(GeneticsPed)

ped2 <-Pedigree(dat.hb,ascendant=c("fatherid","motherid"),ascendantSex = c("M","F"))

inbreeding(ped2,method = "tabular",sort=FALSE)
```

A waste of time - it just gives the diagonal of the kinship matrix, and a somewhat garbled one at that.

### Exam question

Taken from pierce test bank.

```{r}
randomNames(40,gender="Male",which.names="first")
```

```{r}
drawped("ExamQuestion")
```

